src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -82,13 +82,35 @@ public static function fromDsn(string $dsn, array $options = array(), bool $debu
         }
 
         $exchangeOptions = $amqpOptions['exchange'];
-        $queueOptions = $amqpOptions['queue'];
+        $queueOptions = self::getQueueOptions($amqpOptions['queue']);
 
         unset($amqpOptions['queue'], $amqpOptions['exchange']);
 
         return new self($amqpOptions, $exchangeOptions, $queueOptions, $debug, $amqpFactory);
     }
 
+    private static function getQueueOptions(array $queueOptions): array
+    {
+        if (!array_key_exists('arguments', $queueOptions)) {
+            return $queueOptions;
+        }
+
+        foreach ($queueOptions['arguments'] as $key => $value) {
+            switch ($key) {
+                case 'x-message-ttl':
+                case 'x-max-length':
+                case 'x-max-length-bytes':
+                case 'x-max-priority':
+                    // RabbitMQ expects the arguments above to be passed as integer values
```

Comment:
Should we place this comment/behavior in this class? RabbitMQ is one of many message brokers the community are going work with.

Reply:
These are the four parameters that I mentioned in the [original issue](https://github.com/symfony/symfony/issues/29044). I'm not sure though if these four are all possible parameters that are expected to be an integer, and apart from that they seem to be RabbitMQ-specific or even extensions (`x-`).

I wonder if there is a more generic solution to this problem.

Reply:
From what I see on the [doc][0] there is at least `x-expires` and `x-delay` [there][1] where both are integers.

For sure something more generic should be better. Doing so maybe using a customizable schema as arguments are closed to the server implementation and its extensions.

[0]: https://www.rabbitmq.com/ttl.html#queue-ttl
[1]: https://www.rabbitmq.com/blog/2015/04/16/scheduling-messages-with-rabbitmq/

Reply:
:+1: